public/app/app.ts

```diff
@@ -150,6 +151,8 @@ export class GrafanaApp {
       configureStore();
       initExtensions();
 
+      initAlerting();
```

Comment:
I don't think we want to do this on every page, slows down the initial load & startup for everything

Reply:
any way to do this lazy (on demand when needed)

Reply:
How else can we use the `addCustomRightAction` function from Alerting? 
`initAlerting` doesn't wait for rules to be loaded. It only adds the toolbar button

Reply:
I think this is fine actually. Was thinking initially we could have that action added directly in the component rendering the toolbar, but this seems more reasonable as it decouples alerting from dashboarding. 

I only suggest to rename this function to be `initAlertingToolbarActions`.

Reply:
The `initAlerting` function only initializes the `addCustomRightAction`. Then the button lazily calls the Alerting API to figure out whether it should render. 
The `initAlerting` call shouldn't impact the initial load in any way

Reply:
@dprokop how is this fine? does not work around all the webpack chunking making so we have to load every alerting assets on every page load (before the app is booted) , not matter the route?

Reply:
I didn't think about it. You may be right that we need to React.lazy this component